firebase-firestore-ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx
+
+import android.support.annotation.Keep
+import com.google.firebase.FirebaseApp
+import com.google.firebase.components.Component
+import com.google.firebase.components.ComponentRegistrar
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FieldPath
+import com.google.firebase.firestore.FirebaseFirestore
+import com.google.firebase.firestore.QueryDocumentSnapshot
+import com.google.firebase.firestore.QuerySnapshot
+
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.platforminfo.LibraryVersionComponent
+
+/** Returns the [FirebaseFirestore] instance of the default [FirebaseApp]. */
+val Firebase.firestore
+    get() = FirebaseFirestore.getInstance()
+
+/** Returns the [FirebaseFirestore] instance of a given [FirebaseApp]. */
+fun Firebase.firestore(app: FirebaseApp): FirebaseFirestore = FirebaseFirestore.getInstance(app)
+
+/** Returns the contents of the document converted to a POJO or null if the document doesn't exist. */
+inline fun <reified T> DocumentSnapshot.toObject(): T? = toObject(T::class.java)
+
+/** Returns the contents of the document converted to a POJO or null if the document doesn't exist. */
+inline fun <reified T> DocumentSnapshot.toObject(
+    serverTimestampBehavior: DocumentSnapshot.ServerTimestampBehavior
+): T? =
+        toObject(T::class.java, serverTimestampBehavior)
+
+/**
+ * Returns the value at the field, converted to a POJO, or null if the field or document doesn't
+ * exist.
+ */
+inline fun <reified T : Any> DocumentSnapshot.getField(field: String): T? = get(field, T::class.java)
```

Comment:
`reified T : Any` as a type parameter here seems wrong. If the function can return `null` (because the return type is `T?`) shouldn't this be just `reified T`?

Reply:
Good catch, `Any` was not intentional, just an oversight, removed here and below